### ISSUE TITLE
Normalizer does not work as expected

### ISSUE THREAD

**Issue by nick863** [microsoft/onnxruntime#16451]
### Describe the issue

Create a scikit-learn regression pipeline, consisting from LassoLars learner and Normalizer, which is being normalized by Max
This model can be converted to ONNX using skl2onnx. Make sure to use sklearn later then v. 0.23.0. In this version the divder for normalization is being calculated as maximum of absolute values:

```python
norms = np.max(abs(X), axis=1)
```
But onnxruntime ormalizer still takes a maximal value of a data set: https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/cpu/ml/normalizer.cc:

```C++
float max = std::numeric_limits<float>::lowest();

for (int i = 0; i < batch_size; ++i) {
    max = std::max(max, static_cast<float>(*in++));
}
```
Because of this issue, if max(x) != max(abs(x)), the results will be different. Is it possible to use the absolute value?

### To reproduce

```python
import unittest

import numpy as np
import pandas as pd

from sklearn import datasets
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import Normalizer
from sklearn.neighbors import KNeighborsRegressor
from sklearn.model_selection import train_test_split

from skl2onnx import convert_sklearn
from skl2onnx.common.data_types import FloatTensorType

import onnxruntime as ort


class TestNaiveONNXConversion(unittest.TestCase):
    """Test conversion without AutoML"""

    def test_conversion(self):
        """Do the test."""
        # Generate the input from scikit-learn diabetes data set.
        X, y = datasets.load_diabetes(return_X_y=True)
        x_train, x_test, y_train, _ = train_test_split(X, y, test_size=0.2, random_state=0)
        features = ['age', 'gender', 'bmi', 'bp', 's1', 's2', 's3', 's4', 's5', 's6']
        x_train = pd.DataFrame(data=x_train, columns=features)
        x_test = pd.DataFrame(data=x_test, columns=features)
        # Create and train the model.
        reg = Pipeline(steps=[
            ('SparseNormalizer', Normalizer(norm='max')),
            ('KNeighborsRegressor', KNeighborsRegressor(metric='euclidean', n_jobs=1, n_neighbors=62, p=4))
        ])
        reg.fit(x_train, y_train)
        # Get predictions from scikit-learn.
        predictions = np.round(reg.predict(x_test), decimals=3)
        
        # Convert the model to ONNX
        initial_type = [('float_input', FloatTensorType([None, len(features)]))]
        onx = convert_sklearn(reg, initial_types=initial_type)
        with open("test.onnx", "wb") as f:
            f.write(onx.SerializeToString())
        
        # Load the model on ONNX runtime
        inference_session = ort.InferenceSession("test.onnx")
        
        # Do the inference on ONNX model
        outputs = inference_session.run(None, {"float_input": x_test.values.astype(np.float32)})
        pred_onnx = np.round(np.array(outputs).astype(np.float32), decimals=3).reshape(-1)
        # Print matches/mismatches
        print(np.isclose(pred_onnx, predictions, rtol=1e-2))
        # Finally compare the predictions
        assert np.allclose(pred_onnx, predictions, rtol=1e-2)
        

if __name__ == '__main__':
    unittest.main()
```

### Platform

Windows

### OS Version

11

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

1.14.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

Default CPU
